test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -1431,18 +1436,20 @@ public void help() throws Exception {
     }
   }
 
-  // TODO - evaluate this test is testing what is expected or history is working.
   @Test
   public void history() throws Exception {
-    final String table = name.getMethodName();
+    final String table = getUniqueNames(1)[0];
 
-    ts.exec("history -c", true);
-    ts.exec("createtable " + table);
-    ts.exec("deletetable -f " + table);
-    // TODO - this may be passing only because method name is history.
+    // test clear history command works
+    ts.writeToHistory("foo");
+    ts.exec("history -c", true, "foo", false);
+
+    // write to history file. Does not execute functions
+    ts.writeToHistory("createtable " + table);
+    ts.writeToHistory("deletetable -f " + table);
+
+    // test that history command prints contents of history file
     ts.exec("history", true, table, true);
```

Comment:
This could be more thorough in checking that the createtable and deletetable commands were added:

```suggestion
    ts.exec("history", true, "createtable " + table, true);
    ts.exec("history", true, "deletetable -f " + table, true);
```

Reply:
I agree on this point, I should have done that earlier.